DirectProgramming/DPC++FPGA/include/pipe_utils.hpp

```diff
@@ -0,0 +1,242 @@
+//==============================================================
+// Copyright Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+#ifndef __PIPE_ARRAY_HPP__
+#define __PIPE_ARRAY_HPP__
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/fpga_extensions.hpp>
+#include <utility>
+
+/*
+
+This header defines the following utilities for use with pipes in DPC++ FPGA
+designs.
+
+1. PipeArray
+
+     Create a collection of pipes that can be indexed like an array.
+
+     template <class Id,          // identifier for the pipe array
+               typename BaseTy,   // type to write/read for each pipe
+               size_t min_depth,  // minimum capacity of each pipe
+               size_t... dims     // depth of each dimension in the array
+                                  // any number of dimensions are supported
+               >
+     struct PipeArray
```

Comment:
Maybe we could add a simple usage of pipe array here. E.g., how to declare and get one of the pipes. Like

```c++
class PipeArrayId;
constexpr int min_depth = 0;
constexpr int num_pipes = 4;
using MyPipeArray = PipeArray<PipeArrayId, int, min_depth, num_pipes>;

/* ... */

constexpr int pipe_idx = 1;
MyPipeArray::GetPipe<pipe_idx>::read();
```

Reply:
I'm happy to add examples here. Should we do that for all exposed helper functions/classes in the header?

Reply:
Done for both PipeArray and PipeDuplicator